### ISSUE TITLE
Vim layout is broken on iPadOS 14.5

### ISSUE THREAD

**Issue by kkk669** [holzschu/a-shell#176]
This problem may be fixed before the final build of iPadOS 14.5 if it is caused by OS or SDK, but I created this issue just in case.

In iPadOS 14.5 Beta 1, the layout of Vim is broken on a-Shell. Please see the screenshots below.

1. `vim` with [my .vimrc](https://github.com/kkk669/dotfiles/blob/27b7dedb2319246c4b45143017a16ddb8fe8546b/.vimrc) on a-Shell 1.6.6 (iPadOS 14.5 Beta 1):

![IMG_0399](https://user-images.githubusercontent.com/601636/106895138-3380bc80-6733-11eb-98ef-2f5e49c1782f.png)

2. `vim` with [my .vimrc](https://github.com/kkk669/dotfiles/blob/27b7dedb2319246c4b45143017a16ddb8fe8546b/.vimrc) on a-Shell 1.6.6 (iPadOS 14.4):

![IMG_0247](https://user-images.githubusercontent.com/601636/106895175-3ed3e800-6733-11eb-9502-d56cc12ae4e9.png)

3. `vim -u NONE` on a-Shell 1.6.6 (iPadOS 14.5 Beta 1):

![IMG_0400](https://user-images.githubusercontent.com/601636/106895161-3aa7ca80-6733-11eb-8816-6d7c7a3480ea.png)

**Comment by kkk669** [comment]
The environment of the first screenshot had worked normally before I updated iPadOS from 14.4 to 14.5. When it was iPadOS 14.4, it looked exactly the same as the second screenshot.

**Comment by holzschu** [comment]
Thanks for the advance warning. It's a bit weird that the iOS version has an impact here.

**Comment by kkk669** [comment]
iPadOS 14.5 Developer/Public Beta 2 has been released some days ago, but the issue still occurs.

**Comment by kkk669** [comment]
> It's a bit weird that the iOS version has an impact here.

iOS 14.5's Safari brings the new version of WebKit engine, so it might be affecting this issue.

It includes some new features and fixes like below:

https://www.reddit.com/r/ipad/comments/lifj97/webkit_fix_for_the_scrolling_issue_which_landed/

Moreover, I tested iSH with the same .vimrc, and then it was broken too. It uses hterm as a-Shell does, so I believe that it is a bug between WebKit and hterm.

![2338D899-980C-4DC8-B13A-30A4A5C3F7E6](https://user-images.githubusercontent.com/601636/108829330-63710080-760b-11eb-9928-a9259f2d6e7f.png)

**Comment by holzschu** [comment]
Ah! Thanks for the explanation. Do you know if Safari on OSX uses this new version of the WebKit engine? Because if yes, I can test on the simulator.

**Comment by kkk669** [comment]
Thank you for your quick response.

Safari on macOS should be differ from the one on iOS, so I believe it can't reproduce the problem.

However I haven't tested it on the iOS Simulator due to lack of my Mac storage, I believe that you can reproduce it on the Simulator.

**Comment by holzschu** [comment]
Hi, beta 3 is out, do you know if the problem is still present?

**Comment by kkk669** [comment]
Hi @holzschu, I checked right now, but it still occurs on iPadOS 14.5 beta 3.

![IMG_0433](https://user-images.githubusercontent.com/601636/109785148-3053f000-7c4f-11eb-876d-92a1fb898fec.png)

**Comment by holzschu** [comment]
I cloned the entire dotfile directory, but I still have issues when I run vim. Does that ring a bell?
![Simulator Screen Shot - iPad Pro (9 7-inch) - 2021-03-03 at 11 36 25](https://user-images.githubusercontent.com/8070736/109793242-b0a92f80-7c14-11eb-8f91-13bb46a8c1c2.png)

**Comment by kkk669** [comment]
My dotfiles repo is using submodules, so you have to clone it recursively. I often use Working Copy to clone it because it supports submodules.

**Comment by holzschu** [comment]
Got it. It's now running in the simulator. 
Now there is a way to make the simulator use the WebKit version from the nightly build of Safari, if I can find it again.
![Simulator Screen Shot - iPad Pro (9 7-inch) - 2021-03-03 at 11 41 44](https://user-images.githubusercontent.com/8070736/109793820-6eccb900-7c15-11eb-8e8b-2a93d39da938.png)

**Comment by kkk669** [comment]
@holzschu 

> Now there is a way to make the simulator use the WebKit version from the nightly build of Safari, if I can find it again.

How about iPadOS 14.5 simulator? You can download it with Xcode 12.5 beta.

**Comment by holzschu** [comment]
I'm going to give it a try, but in previous versions, the simulator was running iOS beta, but the webkit was from Safari (which makes sense, but is annoying).

**Comment by holzschu** [comment]
Good news: it does the same in the simulator. According to the JS console (available from Safari "Develop" menu), the issue comes from a function named: `splitWidecharString`, when it calls `it.next()`: (`it.next is undefined`).
```
  const segmenter = new Intl.Segmenter(undefined, {type: 'grapheme'});
  const it = segmenter.segment(str);

  const rv = [];
  let segment = it.next();
```

**Comment by holzschu** [comment]
I've pushed a version of `hterm_all.js` that fixes the issue in the simulator. Could you check that it works on your side?

**Comment by kkk669** [comment]
Thanks. I'll try it later.

**Comment by kkk669** [comment]
I'm trying to build cpython to run and check Vim on the latest commit of a-Shell with [your latest script](https://github.com/holzschu/cpython/tree/45c7ea68eb0a73757f1940993c319ee8f1bda4d9). It seems to have some requisites (e.g. `sudo xcode-select --install` and OpenSSL 1.1) that isn't written in README.md.

**Comment by kkk669** [comment]
I've confirmed that the issue has been fixed in 1.6.9. Thank you very much.
